luzer/fuzzed_data_provider.cc

```diff
@@ -208,6 +208,37 @@ luaL_remaining_bytes(lua_State *L)
 	return 1;
 }
 
+/* Returns a random element of specified array. */
+static int
+luaL_oneof(lua_State *L)
+{
+	lua_userdata_t *lfdp;
+	lfdp = (lua_userdata_t *)luaL_checkudata(L, 1, FDP_LUA_UDATA_NAME);
+	luaL_checktype(L, 2, LUA_TTABLE);
+
+	int len = 0;
+	/* Push starting `nil` for iterations. */
+	lua_pushnil(L);
+	while (lua_next(L, 2) != 0) {
+		/*
+		 * Remove `value` from the stack. Keeps `key` for
+		 * the next iteration.
+		 */
+		lua_pop(L, 1);
+		len++;
+	}
+
+	if (len == 0) {
```

Comment:
Side note: I wonder, may it be useful to allow the empty table argument for the guaranteed `nil` to be chosen (if we insert elements in the table conditionally)? What do you think?

Reply:
So, the test from the previous comment should be updated as well, shouldn't it?